instrumentation/opentelemetry-instrumentation-aiohttp-server/src/opentelemetry/instrumentation/aiohttp_server/__init__.py

```diff
@@ -464,6 +493,10 @@ class AioHttpServerInstrumentor(BaseInstrumentor):
     """
 
     def _instrument(self, **kwargs):
+        tracer_provider = kwargs.get("tracer_provider", None)
+        assert tracer_provider is None or isinstance(
```

Comment:
We don't usually add these kind of assertions, maybe you can add some typing instead?